### PR TITLE
Add Kemiq Q2 Beta and Kemiq Q2 Gama definitions, mesh and qualities

### PR DESCRIPTION
**Pull request by kemiq3d**

### CODE REVIEW

**Comment by Ghostkeeper** [comment]
Developers, see issue CURA-4086.

**Comment by kemiq3d** [comment]
Added new lines at the end of the files

File: resources/definitions/kemiq_q2_beta.def.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    "name": "Kemiq Q2 Beta",
+    "inherits": "fdmprinter",
+    "metadata": {
+        "visible": true,
+        "author": "KEMIQ",
+        "manufacturer": "KEMIQ",
+        "category": "Other",
+        "file_formats": "text/x-gcode",
+        "platform": "kemiq_q2.stl",
+		"has_machine_quality": true,
```
Comment (LipuFei) [inline resources/definitions/kemiq_q2_beta.def.json]: Please use `spaces` instead `TABs`, and the same applies to the lower part and `kemiq_q2_gama.def.json`

File: resources/quality/kemiq_q2/kemiq_q2_beta_abs_draft.inst.cfg [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+version = 2
+name = Draft
+definition = kemiq_q2_beta
+
+[metadata]
+type = quality
+weight = -3
+material = generic_abs
+quality_type = coarse
+setting_version = 1
```
Comment (LipuFei) [inline resources/quality/kemiq_q2/kemiq_q2_beta_abs_draft.inst.cfg]: We are now using `setting_version = 2`, please update this for all the quality files.

File: resources/quality/kemiq_q2/kemiq_q2_beta_pla_extra_fine.inst.cfg [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+material_diameter = 1.75
+
+retraction_speed = 40
+retraction_amount = 2.15
+
+raft_airgap = 0.25
+
+
+
+
```
Comment (LipuFei) [inline resources/quality/kemiq_q2/kemiq_q2_beta_pla_extra_fine.inst.cfg]: Those extra empty lines can be removed. The same applies to the other 2 files.

File: resources/quality/kemiq_q2/kemiq_q2_gama_pla_extra_fine.inst.cfg [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+material_print_temperature = 214
+material_diameter = 1.75
+
+retraction_speed = 40
+retraction_amount = 2.15
+
+raft_airgap = 0.25
+
+
```
Comment (LipuFei) [inline resources/quality/kemiq_q2/kemiq_q2_gama_pla_extra_fine.inst.cfg]: Extra empty lines can be removed

File: resources/quality/kemiq_q2/kemiq_q2_gama_pla_normal.inst.cfg [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+material_print_temperature = 214
+material_diameter = 1.75
+
+retraction_speed = 40
+retraction_amount = 2.15
+
+raft_airgap = 0.25
+
+
```
Comment (LipuFei) [inline resources/quality/kemiq_q2/kemiq_q2_gama_pla_normal.inst.cfg]: Extra empty lines can be removed

**Comment by alekseisasin** [comment]
We will fix rest remarks for you.

**Comment by kemiq3d** [comment]
No problem,
we will correct the issues as you say and also will do a few changes on the quality files till tomorrow.

Thank you.

File: resources/definitions/kemiq_q2_beta.def.json (hunk 1 above)
Comment (kemiq3d) [inline resources/definitions/kemiq_q2_beta.def.json]: We made changes in definition scripts as you said but it failed. Please tell us what we can do.

**Comment by kemiq3d** [comment]
We proposed file changes for all definition and quality files. In the middle of this changes they warned me about not to pull request for each file. And we just proposed and did not pull request for the rest of the files.
But i am wondering if proposing changes enough or do i have to pull a single request for each changed file? Sorry for the mess but i couldn't understand the procedure entirely. 
Thanks for your help.

**Comment by Ghostkeeper** [comment]
It's okay to deliver a bunch of pull requests like this but we'd prefer it if you were to bunch them up a bit. If you need to change a setting in all of Kemiq3D's profiles then it makes sense to request us to pull in all of those changes, rather than to request us 9 times to pull in a change in one profile. It means a bit less administration from our side.

Normally a pull request goes like this:

* Someone wants to change something in Cura (like changing a bunch of settings in a bunch of profiles).
* He forks Cura. Now he has his own copy of Cura's code that he can make changes to.
* He makes a number of commits in his own fork. Each commit should be one logical change with a purpose, such as "Improve bed adhesion for Kemiq3D" that changes a number of the initial layer settings for all of the Kemiq3D profiles.
* When he's done with the changes, he makes a pull request in the main Cura repository. That is, he requests us to merge his changes into the main distribution. Just one pull request for the entire feature/improvement/bugfix.
* The Cura developers review the changes, test it where necessary and merge the code if everything is all right.

This original pull request that we're commenting on seems to have worked like that. It was an excellent pull request.

**Comment by kemiq3d** [comment]
Thank you very much for this explanatory answer.

**Pull request merged**